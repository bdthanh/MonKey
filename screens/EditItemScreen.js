import React, {useState, useEffect} from 'react';
import { View, Text, TouchableOpacity,  Modal, TextInput, Alert, Pressable, Keyboard, StyleSheet, FlatList} from 'react-native';
import { colors } from '../components/colors';
import moment from 'moment';
import DateTimePicker from '@react-native-community/datetimepicker';
import { collection, onSnapshot, deleteDoc, doc, getDoc, setDoc, updateDoc, getDocs, query } from 'firebase/firestore';
import { db } from '../utils/db';
import { ScreenWidth } from '../components/constants';
import { getUserID } from '../utils/authentication'; 
import { StatusBarHeight } from '../components/constants';
import { StatusBar } from 'expo-status-bar';
import { MaterialCommunityIcons, Entypo, Foundation } from '@expo/vector-icons'
import { Timestamp } from 'firebase/firestore';
import CurrencyInput from 'react-native-currency-input';
import { useFocusEffect } from '@react-navigation/native';
const { lightBlue, darkBlue, darkYellow,lighterBlue } = colors

const EditItemScreen = ({route, navigation}) => {
  const [ExpenseCategory, setExpenseCategory] = useState([])
  const [IncomeCategory, setIncomeCategory] = useState([])

  const getExpenseCategories = async () => {
    try {
      const expenseCategoryRef = query(collection(db, 'Input Category/Expense/' + getUserID()))
      const expenseCategories = [];
      const expenseCats = await getDocs(expenseCategoryRef)
      expenseCats.docs.forEach((doc) => {
        if (doc.data().name != 'Deleted Category' && doc.data().name != 'Edit') {
          expenseCategories.push({
            name: doc.data().name,
            color: doc.data().color,
            icon: doc.data().icon,
          });
        }
      });
      expenseCategories.sort((a, b) => a.name == 'Edit' ? 1 : b.name == 'Edit' ? -1 : a.name > b.name ? 1 : -1)
      setExpenseCategory(expenseCategories)

    } catch {
      (error) => console.log(error.message)
    }
  }

  const getIncomeCategories = async () => {
    try {
      const incomeCategoryRef = query(collection(db, 'Input Category/Income/' + getUserID()))
      const incomeCategories = [];
      const incomeCats = await getDocs(incomeCategoryRef)
      incomeCats.docs.forEach((doc) => {
        if (doc.data().name != 'Deleted Category' && doc.data().name != 'Edit') {
          incomeCategories.push({
            name: doc.data().name,
            color: doc.data().color,
            icon: doc.data().icon,
          });
        }
      });
      incomeCategories.sort((a, b) => a.name == 'Edit' ? 1 : b.name == 'Edit' ? -1 : a.name > b.name ? 1 : -1)
      setIncomeCategory(incomeCategories)
    } catch {
      (error) => console.log(error.message)
    }
  }

  const [inprogressAmount, setInprogressAmount] = useState('');
  const [inprogressNote, setInprogressNote] = useState('');
  const [inprogressCategory, setInprogressCategory]= useState('');
  let [inprogressDate, setInprogressDate] = useState(moment());
  const [inprogressId, setInprogressId] = useState('');
  const [inprogressType, setInprogressType] = useState('')
  const [colorC, setColor] = useState('')

  useFocusEffect(
    React.useCallback(() => {
      getExpenseCategories()
      getIncomeCategories()
      setInprogressAmount(route.params.inprogressAmount)
      setInprogressNote(route.params.inprogressNote)
      setInprogressCategory(route.params.inprogressCategory)
      setInprogressDate(moment(route.params.inprogressDate, 'YYYY-MM-DD'))
      setInprogressId(route.params.inprogressId)
      setInprogressType(route.params.inprogressType)
      setColor(route.params.color)
      return () => {}
    }, [])
  );
  

  /********** Date Picker Variables **********/
  const [show, setShow] = useState(false);

  const onChange = (event, selectedDate) => {
    setShow(Platform.OS === 'ios');
    setInprogressDate(moment(selectedDate))
  }

  const addOneDay = () => {
    inprogressDate = setInprogressDate(moment(inprogressDate).add(1, 'day'));
  }

  const subtractOneDay = () => {
    inprogressDate = setInprogressDate(moment(inprogressDate).subtract(1, 'day'));
  }
  
  /*************** Function to edit record ***************/
  const editRow = (id, type) => {
    const path = 'Finance/' + getUserID() + '/' + (type == 'income' ? 'Income' : 'Expense')
    const catRef = doc(db, path, id)
    const amountNumber = Number(inprogressAmount)
      updateDoc(catRef, {
        date: inprogressDate.format('DD'),
        month: inprogressDate.format('MM'), 
        year: inprogressDate.format('YYYY'), 
        amount: amountNumber,
        note: inprogressNote,
        category: inprogressCategory,
        categoryIcon: type == 'income' ? IncomeCategory.find(cat => cat.name == inprogressCategory).icon : ExpenseCategory.find(cat => cat.name == inprogressCategory).icon,
        categoryColor: type == 'income' ? IncomeCategory.find(cat => cat.name == inprogressCategory).color : ExpenseCategory.find(cat => cat.name == inprogressCategory).color,
        notedAt: Timestamp.now(), 
      })
  }
  const closeEditModal = () => {
    setInprogressCategory('')
    setInprogressNote('')
    setInprogressAmount('')
    setInprogressDate(moment())
    setInprogressId('')  
    setInprogressType('')
    setShow(false)
    navigation.goBack()
    //console.log(inprogressDate.format('DD_MM_YYYY'))
  }
  const onSubmitEdit = () => {
    if (inprogressAmount == null || inprogressAmount == 0) {
      Alert.alert("Alert", "Please enter the amount", [
        {text: 'Okay', onPress: () => console.log('Alert closed')}
      ]);
    }
    else if (inprogressCategory == '') {
      Alert.alert("Alert", "Please choose the category", [
        {text: 'Okay', onPress: () => console.log('Alert closed')}
      ]);
    }
    else if (inprogressAmount != '' && inprogressCategory != '') {
      editRow(inprogressId, inprogressType)
      setInprogressCategory('')
      setInprogressNote('')
      setInprogressAmount('')
      setInprogressDate(moment())
      setInprogressId('')
      setInprogressType('')
      setShow(false)
      navigation.goBack();
    } 
    
  }

  return (
    <View style={styles.container}>
      <StatusBar style='dark'/>
      <Pressable onPress={Keyboard.dismiss}>
        {/*********** Header ***********/}
        <View style={[styles.headerModal, {flexDirection:'row'}]}>
          <View style={{flex:4}}>
            <Text></Text>
          </View>
          <View style={{flex:8, alignItems:'center',justifyContent:'center'}}>
            <Text style={styles.boldBlueHeaderText}>Edit item</Text>
          </View>
          <View style={{flex:4, alignItems:'center', justifyContent:'center', marginBottom:10}}>
            <TouchableOpacity 
              style={{alignItems:'center', justifyContent:'center',backgroundColor:darkYellow,height:30,width:65, borderRadius:5}}
              onPress={closeEditModal}>
              <Text style={styles.cancelText}>Cancel</Text>
            </TouchableOpacity>
          </View>
        </View>

        {/*********** Type ***********/}
        <View style={styles.noteView}>
          <View style={{
            flex:20,
            paddingLeft:12,
            justifyContent:'center'
            }}>
            <Text style={styles.dateText}>Type</Text>
          </View>
          <View style={{
            flex:80,
            alignItems:'center',
            justifyContent:'center',
            borderBottomColor:darkYellow,
          }}>
            <View 
              style={{alignItems:'center',justifyContent:'center', backgroundColor:inprogressType=='income' ? '#e2f5e2' : '#fdddcf', width:210, height:34, borderRadius:20}}
              >
              <Text style={{color: inprogressType=='income' ? '#26b522' : '#ef5011', fontSize:18, fontWeight:'500'}}>
                {inprogressType=='income' ? 'Income' : 'Expense'}
              </Text>
            </View>
          </View>
        </View>

        {/*********** Date ***********/}
        <View style={styles.dateView}>
          <View style={{
            flex:20,
            paddingLeft:12,
            justifyContent:'center'
          }}>
            <Text style={styles.dateText}>Date</Text>
          </View>
          <View style={{
            flex:15,
            alignItems:'center',
            justifyContent:'center'
          }}>
            <TouchableOpacity style={{position: 'absolute'}} onPress={subtractOneDay}>
              <Entypo name='chevron-left' size={28} color={darkBlue}/>
            </TouchableOpacity>
          </View>
          <View style={styles.datePickerView}>
            <TouchableOpacity onPress={()=>setShow(true)}>
              <View>
                <Text style={styles.dateText}>{inprogressDate.format('MMMM Do, YYYY')}</Text>
              </View>
            </TouchableOpacity>
          </View>
          <View style={{
            flex:15,
            alignItems:'center',
            justifyContent:'center'
          }}>
            <TouchableOpacity style={{position: 'absolute'}} onPress={addOneDay}>
              <Entypo name='chevron-right' size={28} color={darkBlue}/>
            </TouchableOpacity>
          </View>
        </View>
        <Modal 
          visible={show}
          transparent={true}
          animationType='fade'
          style={{backgroundColor:'#fff'}}
        >
          <View style={styles.modalView}>
            <View style={{backgroundColor:'#fff', width:'100%', paddingVertical:15, borderTopRightRadius:20, borderTopLeftRadius:20}}>
              <View style={{flexDirection:'row', alignItems:'center', justifyContent:'center', paddingTop:10, paddingLeft:20, paddingRight:20}}>
                <View style={{flex:5}}>
                  <TouchableOpacity onPress={()=> {setShow(false), setInprogressDate(moment())}}>
                    <View>
                      <Text style={styles.datePickerOffText}>Cancel</Text>
                    </View>
                  </TouchableOpacity>
                </View>
                <View style={{flex:5, alignItems:'flex-end'}}>
                  <TouchableOpacity onPress={()=> setShow(false)}>
                    <View>
                      <Text style={styles.datePickerOffText}>Done</Text>
                    </View>
                  </TouchableOpacity>
                </View>
              </View>
              <View style={{
                borderBottomColor: '#E9E9E9',
              }}>
                <DateTimePicker
                  value={new Date(inprogressDate)}
                  is24Hour={true}
                  textColor={darkBlue}
                  display='spinner'
                  onChange ={onChange}
                  minimumDate={new Date(moment().subtract(50, 'years').format('YYYY-MM-DD'))}
                  maximumDate={new Date(moment().add(50, 'years').format('YYYY-MM-DD'))}
                />
              </View>
            </View>
          </View>
        </Modal>

        {/*********** Amount ***********/}
        <View style={styles.dateView}>
          <View style={{
            flex:30,
            paddingLeft:12,
            justifyContent:'center'
            }}>
            <Text style={styles.dateText}>{inprogressType == 'income' ?'Income' : 'Expense'}</Text>
          </View>
          <View style={{flex:5}}></View>
          <View style={styles.datePickerView}>
            <CurrencyInput
              style={[styles.inputContainer, {textAlign:'right'}]}
              value={inprogressAmount}
              onChangeValue={setInprogressAmount}
              delimiter=","
              separator="."
              precision={2}
              placeholder='0.00'
              maxLength={16}
              keyboardType='decimal-pad'
              placeholderTextColor={lightBlue}
            />
          </View>
          <View style={{
            flex:15, 
            justifyContent:'center',
            alignItems:'center'
            }}>
              <Foundation name='dollar' size={34} color={darkBlue}/>
          </View>
        </View>
        {/*********** Note ***********/}
        <View style={styles.noteView}>
          <View style={{
            flex:20,
            paddingLeft:12,
            justifyContent:'center'
            }}>
            <Text style={styles.dateText}>Note</Text>
          </View>
          <View style={{
            flex:80,
            alignItems:'center',
            justifyContent:'center',
          }}>
            <TextInput
              maxLength={21}
              multiline={true}
              scrollEnabled={true}
              style={[styles.noteInputContainer, {textAlign:'left'}]}
              placeholder='Note (optional)'
              placeholderTextColor={'#49494950'}
              value={inprogressNote}
              onChangeText={(value) => setInprogressNote(value)}
            />
          </View>
        </View>
        {/*********** Category ***********/}
        <View style={styles.noteView}>
          <View style={{
            flex:22,
            paddingLeft:12,
            justifyContent:'center'
            }}>
            <Text style={styles.dateText}>Category</Text>
          </View>
          <View style={{
              flex:80,
              alignItems:'center',
              justifyContent:'center',
              borderBottomColor:darkYellow,
            }}>
              <Text style={[styles.categoryText, {color:colorC}]}>{inprogressCategory}</Text>
          </View>
        </View>
        <View style={{height:280, alignItems:'center'}}>
          <FlatList
            scrollEnabled={true}
            contentContainerStyle={{alignSelf: 'flex-start'}}
            numColumns={3}
            showsHorizontalScrollIndicator={false}
            showsVerticalScrollIndicator={true}
            data={inprogressType == 'income' ? IncomeCategory : ExpenseCategory} 
            renderItem={({item}) => {
              if (!item.isEdit) {
                return (
                  <View style={styles.itemView}>
                    <TouchableOpacity 
                      style={styles.itemButton}
                      onPress={() => {setInprogressCategory(item.name); setColor(item.color)}}>
                      <MaterialCommunityIcons name={item.icon} size={20} color={item.color}/>
                      <Text style={[styles.categoryButtonText]}>{' ' + item.name}</Text>
                    </TouchableOpacity>
                  </View>
                )
              }
            }}
          />
        </View>
        {/*********** Submit button ***********/}
        <View style={[styles.submitButtonView, {alignItems:'center', justifyContent:'center', }]}>
          <TouchableOpacity 
          style={[styles.inputButton]} 
          onPress={() => {onSubmitEdit()}}> 
          {/*********** modify function onSubmitEdit and editRow (Ctrl F) ***********/}
            <Text style={styles.cancelText}>Submit</Text>
          </TouchableOpacity>
        </View>
      </Pressable>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex:1,
    backgroundColor: '#fff',
  },
  boldBlueHeaderText: {
    fontSize: 35,
    color: darkBlue,
    marginBottom: 10,
    fontWeight: 'bold',
  },
  categoryText: {
    fontSize:20,
    fontWeight:'bold'
  },
  noteText: {
    fontSize:15,
    fontWeight:'400'
  },
  amountText: {
    fontSize:20,
    fontWeight:'bold'
  },
  headerModal: {
    alignItems:'flex-end', 
    justifyContent:'center',
    backgroundColor:'#fff',
    borderBottomColor:'#808080',
    borderBottomWidth:1,
    height: StatusBarHeight + 42,
  },
  cancelText: {
    fontSize:18,
    color:'#fff',
    fontWeight:'500',
  },
  inputButton: {
    padding: 5,
    flexDirection: 'row',
    height: 40,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius:10,
    backgroundColor:darkBlue,
    width:200,
  },
  propertiesView: {
    flexDirection:'row',
    paddingBottom:7,
    paddingTop:7,
    paddingLeft:4,
    borderBottomColor: '#E9E9E9',
    borderTopColor: '#E9E9E9',
    borderTopWidth:1,            
    height:48
  },
  dateView: {
    flexDirection:'row',
    paddingBottom:7,
    paddingTop:7,
    paddingLeft:4,
    borderBottomColor: '#E9E9E9',
    borderTopColor: '#E9E9E9',
    borderTopWidth:1,            
    height:48
  },
  dateText: {
    fontSize: 19,
    fontWeight: '500',
    color: '#494949',
  },
  datePickerOffText: {
    fontSize: 18,
    fontWeight: '400',
    color: lightBlue,
  },
  datePickerView: {
    flex:55,
    alignItems:'center',
    justifyContent:'center',
    backgroundColor: lighterBlue,
    borderRadius:10
  },
  inputContainer: {
    backgroundColor: lighterBlue,
    color: '#494949',
    borderColor: darkBlue,
    paddingRight: 12,
    width:200,
    borderRadius: 10,
    fontSize: 20,
    fontWeight:'600',
  },
  noteView: {
    flexDirection:'row',
    paddingBottom:4,
    paddingTop:4,
    paddingLeft:4,
    borderBottomColor: '#E9E9E9',
    borderTopColor: '#E9E9E9',  
    borderTopWidth:1,      
    //borderBottomWidth:1,    
    height:48
  },
  noteInputContainer: {
    color:'#494949',
    backgroundColor: '#fff',
    paddingRight: 12,
    width:200,
    borderRadius: 10,
    fontSize: 17,
    fontWeight:'500',
    height: 36,
    paddingHorizontal: 5
  },
  categoryText: {
    fontSize: 18,
    fontWeight: '500',
  },
  itemView: {
    flexDirection:'row',
    alignItems:'center',
    justifyContent:'center',
    height: 50,
    width:(ScreenWidth-15)/3,
    marginTop: 20 
  },
  itemButton: {
    flexDirection: 'column',
    height: 60,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius:10,
    backgroundColor:'#fff',
    width:ScreenWidth/3-15,
    shadowColor:'#999',
    shadowOffset: {width:0,height:1},
    shadowOpacity:0.8,
    shadowRadius:2,
    marginTop: -20,
    paddingLeft: 5,
    paddingRight: 5
  },
  categoryButtonText: {
    fontSize: 16,
    fontWeight: '300',
  },
  submitButtonView: {
    flexDirection:'row',
    paddingBottom:4,
    paddingTop:4,
    paddingLeft:4,
    borderBottomColor: '#E9E9E9',
    borderTopColor: '#E9E9E9',     
    height:48  
  },
  submitButton: {
    padding: 5,
    height: 40,
    width:100,
    justifyContent: 'center',
    alignItems: 'center',
    borderBottomLeftRadius:10, 
    borderTopLeftRadius:10, 
    borderBottomRightRadius:10,
    borderTopRightRadius:10,
    backgroundColor:lighterBlue
  },
  modalView: {
    flex:1, 
    justifyContent:'flex-end', 
    backgroundColor:'#000000aa',
    shadowOffset: {width:0,height:1},
    shadowOpacity:0.8,
    shadowRadius:2,
    elevation:5,
  }
})

export default EditItemScreen;