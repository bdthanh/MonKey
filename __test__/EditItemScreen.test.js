import React from 'react';
import EditItemScreen from '../screens/EditItemScreen';
import { NavigationContainer } from '@react-navigation/native';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <NavigationContainer>
        <EditItemScreen/>
      </NavigationContainer>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
