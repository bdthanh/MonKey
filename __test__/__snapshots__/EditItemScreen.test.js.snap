// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "backgroundColor": "#fff",
            "borderBottomColor": "#808080",
            "borderBottomWidth": 1,
            "height": 62,
            "justifyContent": "center",
          },
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 4,
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 8,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#35baf6",
              "fontSize": 35,
              "fontWeight": "bold",
              "marginBottom": 10,
            }
          }
        >
          Edit item
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 4,
            "justifyContent": "center",
            "marginBottom": 10,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FAB00E",
              "borderRadius": 5,
              "height": 30,
              "justifyContent": "center",
              "opacity": 1,
              "width": 65,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "500",
              }
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#E9E9E9",
          "borderTopColor": "#E9E9E9",
          "borderTopWidth": 1,
          "flexDirection": "row",
          "height": 48,
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingTop": 4,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 20,
            "justifyContent": "center",
            "paddingLeft": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#494949",
              "fontSize": 19,
              "fontWeight": "500",
            }
          }
        >
          Type
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "#FAB00E",
            "flex": 80,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fdddcf",
              "borderRadius": 20,
              "height": 34,
              "justifyContent": "center",
              "width": 210,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ef5011",
                "fontSize": 18,
                "fontWeight": "500",
              }
            }
          >
            Expense
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#E9E9E9",
          "borderTopColor": "#E9E9E9",
          "borderTopWidth": 1,
          "flexDirection": "row",
          "height": 48,
          "paddingBottom": 7,
          "paddingLeft": 4,
          "paddingTop": 7,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 20,
            "justifyContent": "center",
            "paddingLeft": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#494949",
              "fontSize": 19,
              "fontWeight": "500",
            }
          }
        >
          Date
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 15,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "absolute",
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#cce9f6",
            "borderRadius": 10,
            "flex": 55,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#494949",
                  "fontSize": 19,
                  "fontWeight": "500",
                }
              }
            >
              July 20th, 2022
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 15,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "absolute",
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      style={
        Object {
          "backgroundColor": "#fff",
        }
      }
      transparent={true}
      visible={false}
    />
    <View
      style={
        Object {
          "borderBottomColor": "#E9E9E9",
          "borderTopColor": "#E9E9E9",
          "borderTopWidth": 1,
          "flexDirection": "row",
          "height": 48,
          "paddingBottom": 7,
          "paddingLeft": 4,
          "paddingTop": 7,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 30,
            "justifyContent": "center",
            "paddingLeft": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#494949",
              "fontSize": 19,
              "fontWeight": "500",
            }
          }
        >
          Expense
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 5,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#cce9f6",
            "borderRadius": 10,
            "flex": 55,
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          keyboardType="decimal-pad"
          maxLength={16}
          onChangeText={[Function]}
          placeholder="0.00"
          placeholderTextColor="#ADD8E6"
          style={
            Array [
              Object {
                "backgroundColor": "#cce9f6",
                "borderColor": "#35baf6",
                "borderRadius": 10,
                "color": "#494949",
                "fontSize": 20,
                "fontWeight": "600",
                "paddingRight": 12,
                "width": 200,
              },
              Object {
                "textAlign": "right",
              },
            ]
          }
          value=""
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 15,
            "justifyContent": "center",
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#E9E9E9",
          "borderTopColor": "#E9E9E9",
          "borderTopWidth": 1,
          "flexDirection": "row",
          "height": 48,
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingTop": 4,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 20,
            "justifyContent": "center",
            "paddingLeft": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#494949",
              "fontSize": 19,
              "fontWeight": "500",
            }
          }
        >
          Note
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 80,
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          multiline={true}
          onChangeText={[Function]}
          placeholder="Note (optional)"
          placeholderTextColor="#49494950"
          scrollEnabled={true}
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 10,
                "color": "#494949",
                "fontSize": 17,
                "fontWeight": "500",
                "height": 36,
                "paddingHorizontal": 5,
                "paddingRight": 12,
                "width": 200,
              },
              Object {
                "textAlign": "left",
              },
            ]
          }
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#E9E9E9",
          "borderTopColor": "#E9E9E9",
          "borderTopWidth": 1,
          "flexDirection": "row",
          "height": 48,
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingTop": 4,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 22,
            "justifyContent": "center",
            "paddingLeft": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#494949",
              "fontSize": 19,
              "fontWeight": "500",
            }
          }
        >
          Category
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "#FAB00E",
            "flex": 80,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "fontWeight": "500",
              },
              Object {
                "color": "",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 280,
        }
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "alignSelf": "flex-start",
          }
        }
        data={Array []}
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEnabled={true}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={true}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <View />
      </RCTScrollView>
    </View>
    <View
      style={
        Array [
          Object {
            "borderBottomColor": "#E9E9E9",
            "borderTopColor": "#E9E9E9",
            "flexDirection": "row",
            "height": 48,
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingTop": 4,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#35baf6",
            "borderRadius": 10,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 5,
            "width": 200,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "500",
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</View>
`;
